### PR TITLE
Improve WebXR Layers example

### PR DESCRIPTION
**Pull request by sigmaxipi**
Improve WebXR Layers example.
* Add 4 eye charts to demonstrate the effects of mipmaps on text.
* Add a GUI to demonstrate interacting with layers. (Mentioned in https://github.com/mrdoob/three.js/issues/22120)
* Improve info text and add more comments.
* Add error if layers fail to initialize.
* Add a preview screenshot.

Tested:
* https://raw.githack.com/sigmaxipi/three.js/dev/examples/webxr_vr_layers.html
* Oculus Quest 2 shows layers properly.
* Oculus Go shows error message.

### CODE REVIEW

**Comment by sigmaxipi** [comment]
I'm not sure how to fix the screenshot test. It's failing because the screenshotting system doesn't use layers to render the video or eye charts. I manually took a screenshot which is more informative than the automatic screenshot.

Should I just use the automatic screenshot?

File: examples/webxr_vr_layers.html [hunk 1]
```diff
@@ -185,49 +323,125 @@
...
+
+				// Same as above, but guiLayer.needsUpdate is set when the user interacts with the GUI.
+				if ( guiLayer && ( guiLayer.needsRedraw || guiLayer.needsUpdate ) ) {
+
+					const glBinding = new XRWebGLBinding( session, gl );
+					const glayer = glBinding.getSubImage( guiLayer, frame );
+					renderer.state.bindTexture( gl.TEXTURE_2D, glayer.colorTexture );
+					gl.pixelStorei( gl.UNPACK_FLIP_Y_WEBGL, true );
+					const canvas = guiMesh.material.map.image;
+					gl.texSubImage2D( gl.TEXTURE_2D, 0, 0, 0, canvas.width, canvas.height, gl.RGBA, gl.UNSIGNED_BYTE, canvas );
```
Comment (Mugen87) [inline examples/webxr_vr_layers.html:436]: Sorry but I think all of this overcomplicates the example. We should try to keep it as simple as possible.

Comment (sigmaxipi) [reply examples/webxr_vr_layers.html:436]: Would it be clearer if I pulled the boilerplate code into a utility library at examples/jsm/webxr/layers.js? There is no way to demonstrate text rendering to layers without the `gl.texSubImage2D` call. And one of the major use cases of layers is for high quality text and 2D UI rendering.

Comment (Mugen87) [reply examples/webxr_vr_layers.html:436]: > Would it be clearer if I pulled the boilerplate code into a utility library at examples/jsm/webxr/layers.js?

That sounds like a step in the right direction.

**Comment by mrdoob** [comment]
I agree that it over complicates the example, but I think it's better than the current exmaple.

**Comment by mrdoob** [comment]
However, we should definitely make an abstraction of all this code in `examples/jsm/webxr/` 👍

**Comment by sigmaxipi** [comment]
I'm still working on trying to cleanly abstract the layer code into a separate file. I'm working on my own layer utility, but it's too complex to add into the examples. Once I'm satisfied with the design of my utility lib, I'll extract the minimal bits and create a new PR to update this example.

**Pull request merged**